java/core/src/main/java/com/lancedb/lance/schema/ColumnAlteration.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.lancedb.lance.schema;
+
+import org.apache.arrow.vector.types.pojo.ArrowType;
+
+import java.util.Optional;
+
+/** Column alteration used to alter dataset columns. */
+public class ColumnAlteration {
+
+  private String path;
```

Comment:
`path` was a concept of arrow, should we name it as `column name` for java moudle?

Reply:
IMO, this suggestion makes sense in some dimension.

But, this issue(the ambiguity of concepts in different contexts) also happens in the rust module. This class just aligns with the definition in the rust module.

I am OK, if we all agree with changing the naming in all modules.